assets/javascripts/discourse/initializers/hook-gather-keys.js.es6

```diff
@@ -38,17 +40,21 @@ export default {
     // Hook `Notification` adapter to gather encrypted topic keys.
     NotificationAdapter.reopen({
       find() {
+        const promises = [];
         return this._super(...arguments).then(result => {
           result.notifications.forEach(notification => {
-            if (notification.topic_key) {
-              putTopicKey(notification.topic_id, notification.topic_key);
-              putTopicTitle(
-                notification.topic_id,
-                notification.encrypted_title
-              );
-            }
+            if (!notification.topic_key) return;
+
+            putTopicKey(notification.topic_id, notification.topic_key);
+            putTopicTitle(notification.topic_id, notification.encrypted_title);
+
+            promises.push(
```

Comment:
Move the `const promises = ` into the function, right above the loop over `result.notifications`.